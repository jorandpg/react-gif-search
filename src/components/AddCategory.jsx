import { useState } from "react"
import PropTypes from 'prop-types'

export function AddCategory({onNewValue}) {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        // Seteamos el valor del estado de inputValue, 
        // esto se verÃ¡ reflejado en el input
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length <= 1) return
        onNewValue(inputValue.trim());
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewValue: PropTypes.func.isRequired
}
