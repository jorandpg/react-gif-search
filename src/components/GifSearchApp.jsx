import { useState } from "react"
import {AddCategory, GifGrid} from "./index";

export function GifSearchApp() {

    const [categories, setCategories] = useState(['power'])

    const onAddCategory = (newValue) => {
        if(categories.includes(newValue)) return;
        setCategories([ newValue, ...categories ]);
    }

    return (
        <>  
            <h1>Git Search App</h1>
            
            <AddCategory onNewValue={ onAddCategory } />
          
            {
                categories.map( category => 
                    <GifGrid 
                        key={ category } 
                        category={ category } /> 
                )
            }
        </>
    )
}
