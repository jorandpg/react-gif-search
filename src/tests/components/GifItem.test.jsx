import { render, screen } from "@testing-library/react"
import { GifItem } from "../../components/GifItem"

describe('Pruebas en <GifItem />', () => {

    const title = "Camaleon";
    const url = "https://test.com/";

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={ title } url={ url } />);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
    });

    test('debe mostrar el titulo en el componente', () => {
        render(<GifItem title={ title } url={ url } />);
        
        // Verificamos que el titulo exista
        expect( screen.getByText(title) ).toBeTruthy();
    });
})