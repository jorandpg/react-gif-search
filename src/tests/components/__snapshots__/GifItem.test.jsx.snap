// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Camaleon"
      src="https://test.com/"
    />
    <p>
      Camaleon
    </p>
  </div>
</div>
`;
